tests/posix/timers/src/clock.c

```diff
@@ -29,35 +33,6 @@ static const bool settable[] = {
 	true,
 };
 
-static inline int64_t ts_to_ns(const struct timespec *ts)
-{
-	return ts->tv_sec * NSEC_PER_SEC + ts->tv_nsec;
-}
-
-#define _tp_op(_a, _b, _op) (ts_to_ns(_a) _op ts_to_ns(_b))
-
-#define _decl_op(_type, _name, _op)                                                                \
-	__used static inline _type _name(const struct timespec *_a, const struct timespec *_b)     \
-	{                                                                                          \
-		return _tp_op(_a, _b, _op);                                                        \
-	}
-
-_decl_op(bool, tp_eq, ==);     /* a == b */
-_decl_op(bool, tp_lt, <);      /* a < b */
-_decl_op(bool, tp_gt, >);      /* a > b */
-_decl_op(bool, tp_le, <=);     /* a <= b */
-_decl_op(bool, tp_ge, >=);     /* a >= b */
-_decl_op(int64_t, tp_diff, -); /* a - b */
```

Comment:
Had to remove these to avoid
```
MACRO_WITH_FLOW_CONTROL: Macros with flow control statements should be avoided
```

Reply:
some of these aren't in the `posix_clock.h` header file, causing the compilation error in CI

Reply:
Fixed!